### DISCUSSION TITLE
Creating Popup In Code-behind

### DISCUSSION THREAD

**Discussion by pvogt** [AvaloniaUI/Avalonia#16262] [Q&A] [answered]
### Describe the bug

On the latest Avalonia Nightly Build.

When creating a popup in code-behind, there seems to be an issue in trying to make a popup background transparent. There's a constant white popup background no matter what child controls are used.
This is the popup code I'm using to create it programmatically, which is attached to a Slider to show the timestamp of a media file when mouse is hovering over it:
```
private void CreatePopup()
{
    _popup = new Popup
    {
        PlacementTarget = this,
        IsOpen = false,
        Width = 125
    };
    _popupText = new TextBlock
    {
        Background = Brushes.ForestGreen,
        Foreground = Brushes.Yellow,
    };
    _popup.Child = new Border
    {
        Child = _popupText,
        Background = Brushes.Transparent,
        BorderBrush = Brushes.Transparent,
        HorizontalAlignment = HorizontalAlignment.Center,
    };
}
```
The Width of the popup is set to 125 to show the white box that can't be modified in any way, let alone set to transparent.
What I'm trying to do is to set the child control's (TextBlock) background to transparent and only show the foreground of the child control.
The Background and Foreground colors are set the way it is to make it stand out from the white popup window.
I've attached an image of what it looks like.
![Popup_Issue](https://github.com/AvaloniaUI/Avalonia/assets/1979645/ed7dd589-4d03-4020-8a87-b689a190e496)

### To Reproduce

Create a Popup programmatically using the code I provided.

### Expected behavior

There shouldn't be a white container for the popup if a child control has its background set to transparent.

### Avalonia version

11.2.999-cibuild0049927-alpha

### OS

Windows

### Additional context

Creating a popup in axaml does not produce the white container.
This is what it should look like:
![Popup_Good](https://github.com/AvaloniaUI/Avalonia/assets/1979645/349123db-4b73-48bc-97f1-fefe31af2827)

**Comment by maxkatz6** [comment, marked as answer]
Styling is missing. Even if popup was transparent, all the content would be without any styles applied.
The problem is, the way how you defined Popup doesn't include any Parent information, from which styles can be inherited.
XAML does it automatically, but you need to add something like:
```
this.LogicalChildren.Add(_popup); // where `this` is a Window, or a parent control that hosts popup. 
```

**Reply by pvogt** [reply]
Thank you for the explanation. That fixed my issue.
